ffi/src/slice.rs

```diff
@@ -1,50 +1,62 @@
 //! Logic related to the conversion of slices to and from FFI-compatible representation
 
-use core::mem::MaybeUninit;
+use alloc::{boxed::Box, vec::Vec};
+use core::slice;
 
-use crate::{FfiReturn, OutPtrOf, ReprC, Result};
+use crate::ReprC;
 
-/// Immutable slice with a defined C ABI layout. Consists of data pointer and length
+/// Immutable slice `&[C]` with a defined C ABI layout. Consists of data pointer and length.
+/// `Option<&[C]>` can also be represented by setting the data pointer to a null pointer.
```

Comment:
```suggestion
/// Immutable slice `&[C]` with a defined C ABI layout. Consists of a data pointer and a length.
/// `Option<&[C]>` can also be represented by setting the data pointer to null.
```


or:

```suggestion
/// Immutable slice `&[C]` with a defined C ABI layout. Consists of a data pointer and a length.
/// If the data pointer is set to `null`, the struct represents `Option<&[C]>`.
```


similar changes to other structs below